crates/goose/src/context_mgmt/auto_compact.rs

```diff
@@ -221,8 +238,8 @@ pub async fn check_and_compact_messages(
     Ok(AutoCompactResult {
         compacted: true,
         messages: compacted_messages,
-        tokens_before: Some(tokens_before),
-        tokens_after: Some(tokens_after),
+        tokens_before: Some(tokens_before + SYSTEM_PROMPT_TOKEN_OVERHEAD + TOOLS_TOKEN_OVERHEAD),
+        tokens_after: Some(tokens_after + SYSTEM_PROMPT_TOKEN_OVERHEAD + TOOLS_TOKEN_OVERHEAD),
     })
```

Comment:
This seems wrong; we know how many tokens we had before (sometimes estimated, but for most models we should get a result from them directly). So we also know what the overhead is; the actual tokens before - tokens_before as calculated here. then we can add that to tokens_after for the real value